packages/react/src/components/MultiSelect/FilterableMultiSelect.tsx

```diff
@@ -26,9 +41,179 @@ import { useFeatureFlag } from '../FeatureFlags';
+   * given item to a string label. By default, it extracts the `label` field
+   * from a given item to serve as the item label in the list.
+   */
+  itemToString(item: Item | null): string;
+
+  /**
+   * We try to stay as generic as possible here to allow individuals to pass
+   * in a collection of whatever kind of data structure they prefer
+   */
+  items: Item[];
+
+  /**
+   * Specify the locale of the control. Used for the default `compareItems`
+   * used for sorting the list of items in the control.
+   */
+  locale: string;
+
+  /**
+   * `onChange` is a utility for this controlled component to communicate to a
+   * consuming component what kind of internal state changes are occurring.
+   */
+  onChange?(changes: { selectedItems: Item[] }): void;
+
+  /**
+   * `onInputValueChange` is a utility for this controlled component to communicate to
+   * the currently typed input.
+   *
+   * TODO: Type mismatch between the prop and the Downshift prop.
+   *  This is modelled by the Downshift prop of the same name, but the code
+   *  doesn't pass in the second parameter required by upstreeam type.
+   *  That's why we can't use `DownshiftProps<Item>['onInputValueChange']`
+   */
+  onInputValueChange?(inputValue: string): void;
+
+  /**
+   * `onMenuChange` is a utility for this controlled component to communicate to a
+   * consuming component that the menu was opened(`true`)/closed(`false`).
+   */
+  onMenuChange?(open: boolean): void;
+
+  /**
+   * Initialize the component with an open(`true`)/closed(`false`) menu.
+   */
+  open?: boolean;
+
+  /**
+   * Generic `placeholder` that will be used as the textual representation of
+   * what this field is for
+   */
+  placeholder?: string;
+
+  /**
+   * Specify feedback (mode) of the selection.
+   * `top`: selected item jumps to top
+   * `fixed`: selected item stays at its position
+   * `top-after-reopen`: selected item jump to top after reopen dropdown
+   */
+  selectionFeedback: 'top' | 'fixed' | 'top-after-reopen';
+
+  /**
+   * Specify the size of the ListBox.
+   * Currently, supports either `sm`, `md` or `lg` as an option.
+   */
+  size?: 'sm' | 'md' | 'lg';
+
+  /**
+   * Provide text to be used in a `<label>` element that is tied to the
+   * combobox via ARIA attributes.
+   */
+  titleText?: ReactNode;
+
+  /**
+   * Callback function for translating ListBoxMenuIcon SVG title
+   */
+  translateWithId?(messageId: string, args?: Record<string, unknown>): string;
+
+  type?: 'default' | 'inline';
+
+  /**
+   * Specify title to show title on hover
+   */
+  useTitleInItem?: boolean;
+
+  /**
+   * Specify whether the control is currently in warning state
+   */
+  warn?: boolean;
+
+  /**
+   * Provide the text that is displayed when the control is in warning state
+   */
+  warnText?: ReactNode;
+}
+
+function FilterableMultiSelectRenderFunction<Item extends ItemBase>(
   {
     ['aria-label']: ariaLabel,
+    // @ts-expect-error: deprecated prop
```

Comment:
My thinking was that we shouldn't propagate deprecated props to typescript interfaces, since users shouldn't use them anyway